web/plugin-manager/src/components/pluginBlock.tsx

```diff
@@ -0,0 +1,180 @@
+        }
+        return result;
+    }
+
+    // Launch or stop plugin
+    async function launchAndStopPlugins() {
+        let resultPluginInfo: AxiosResponse<string>;
+        // fetch info from plugin to get fresh data
+        resultPluginInfo = await fetchPluginStatus();
+        // Update data
+        setStatus(isUp(resultPluginInfo.data));
+
+        let result: AxiosResponse<number>;
+        if (toggleStatus) {
+            // Stop plugin
+            try {
+                result = await axiosServices.manageLifePlugins(pluginProperties.id, "stop");
+                forcePlayStatus(false);
+                setStatus(false);
+                return result;
+            } catch (error: any) {
+                sendErrorData("error", `Stop plugin failed , error ${error.message}`);
+            }
+        } else {
+            // Launch plugin
+            try {
+                result = await axiosServices.manageLifePlugins(pluginProperties.id, "start");
+                setStatus(true);
+                forcePlayStatus(true);
+                return result;
+            } catch (error: any) {
+                sendErrorData("error", `Start plugin failed , error :${error.message}`);
+            }
+        }
+    }
+
+    // Update plugin
+    // Not implemented atm
+    function updatePlugins() {
+        //Front end update
+
+        //TODO : Uncoment this when we have a update process
+        //#####################################################
+        // const result = axiosServices.uploadPlugins("filename");
+        // if ( pluginsInfos.status && (pluginsInfos.status <= 200 || pluginsInfos.status >= 300)){
+        //     sendErrorData("error","Plugins infos failed to launch")
+        // }
+        // propsRef.current.online = true
+        // return result;
+        //#####################################################
+        console.log("Update is Not implemented ATM");
+    }
+    // Open plugin homepage
+    function openHomepagePlugins() {
+        if (isUp(pluginProperties.status)) window.open(pluginProperties.home);
+        else {
+            sendErrorData("error", "Plugin is not running can't be launched , launch it first");
+        }
+    }
+    // Uninstall plugin
+    function removePlugins() {
+        try {
+            axiosServices.deletePlugins(pluginProperties.id);
+            p.triggerRefreshPluginList();
+            sendErrorData("success", "Plugin removed");
+        } catch (error:any) {
+            sendErrorData("error", `Plugins failed to be removed , error ${error.message}`);
+        }
+    }
+    //Truncate the string so that it fits in the given lenght if needed.
+    function minimize(str: string, length: number) {
+        if (str.length > length) {
+            return str.substring(0, length) + "...";
+        } else {
+            return str;
+        }
+    }
+
+    // Change the play status icon color and update the status if we want to force it
+    function definePlayStatus() {
```

Comment:
Is it play or  run. I dont understand this play notion. I will call you.

Reply:
play or run was the same but lets go to run